### PR TITLE
feat: generate placeholder dashboard & deploy to github pages

### PR DESCRIPTION
**Pull request by laurentsenta**
Contributes to #123

- [x] Add static site generation with Hugo (go-based, stable, fast, good fit with data templates and taxonomies)
- [x] Update dashboard generation workflow to generate the static hugo site
- [x] Use [Github Workflow Pages](https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/)

> [!IMPORTANT]  
> This is infrastructure work for the generation, the following PRs will munge the data better and output a real, reliable dashboard

- [Example run](https://github.com/singulargarden/gateway-conformance/actions/runs/5936587591)
- [Example output](https://singulargarden.github.io/gateway-conformance/)

### CODE REVIEW

File: .github/workflows/deploy-pages.yml [hunk 1]
```diff
@@ -54,36 +64,70 @@ jobs:
...
     needs: [pull-outputs]
-    # the tests might have failed
-    if: always()
-    defaults:
-      run:
-        shell: bash
     steps:
-      - uses: actions/checkout@v3
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
```
Comment (galargh) [inline .github/workflows/deploy-pages.yml:75]: Let's pin this to sha.

```suggestion
        uses: peaceiris/actions-hugo@16361eb4acea8698b220b76c0d4e84e1fd22c61d # v2.6.0
```

File: .github/workflows/deploy-pages.yml [hunk 2]
```diff
@@ -54,36 +64,70 @@ jobs:
...
+        with:
+          hugo-version: ${{ env.HUGO_VERSION }}
+          extended: true
+      - name: Checkout
+        uses: actions/checkout@v3
         with:
-          path: "gateway-conformance"
+          submodules: recursive
```
Comment (galargh) [inline .github/workflows/deploy-pages.yml:82]: Do we have any submodules to checkout?

```suggestion
```

File: www/config.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+markup:
+  goldmark:
+    renderer:
+      unsafe: true
+
+taxonomies:
+  ipip: ipips
+  test: tests
+  result: results
```
Comment (galargh) [inline www/config.yaml:17]:
```suggestion
  result: results

```

File: www/static/favicon.ico
Comment (galargh) [inline www/static/favicon.ico:1]: Empty file?

File: www/static/logo.png
Comment (galargh) [inline www/static/logo.png:1]: Pretty 😻 What did you use?

File: www/themes/conformance/layouts/shortcodes/gateways-links.html [hunk 4]
```diff
@@ -0,0 +1 @@
+(Gateway Links Work In Progress)
```
Comment (galargh) [inline www/themes/conformance/layouts/shortcodes/gateways-links.html:1]:
```suggestion
(Gateway Links Work In Progress)

```

File: www/themes/conformance/layouts/shortcodes/specs-links.html [hunk 5]
```diff
@@ -0,0 +1 @@
+(Specs Links Work In Progress)
```
Comment (galargh) [inline www/themes/conformance/layouts/shortcodes/specs-links.html:1]:
```suggestion
(Specs Links Work In Progress)

```

File: www/themes/conformance/layouts/shortcodes/url.html [hunk 6]
```diff
@@ -0,0 +1,2 @@
+{{/* make pathing for most URLs stable over any type of domain, base url, etc. */}}
+{{ .Get 0 | absURL }}
```
Comment (galargh) [inline www/themes/conformance/layouts/shortcodes/url.html:2]:
```suggestion
{{ .Get 0 | absURL }}

```

File: www/themes/sk1/LICENSE
Comment (galargh) [inline www/themes/sk1/LICENSE:1]: Is it standard with Hugo to store copies of themes?

File: www/themes/sk1/layouts/partials/header.html [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<!--Site Title-->
+<a href="{{ "" | relLangURL}}">{{site.Title}}</a>
+<!--Sub-title-->
+{{with site.Params.subtitle}} / {{.}}{{end}}
```
Comment (galargh) [inline www/themes/sk1/layouts/partials/header.html:4]:
```suggestion
{{with site.Params.subtitle}} / {{.}}{{end}}

```

File: www/themes/sk1/layouts/partials/footer.html [hunk 8]
```diff
@@ -0,0 +1 @@
+{{with .Site.Copyright}}{{.}} / {{end}}
```
Comment (galargh) [inline www/themes/sk1/layouts/partials/footer.html:1]:
```suggestion
{{with .Site.Copyright}}{{.}} / {{end}}

```

File: www/themes/sk1/layouts/_default/list.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+<ul>
+	{{range $paginator.Pages}}
+	<li>{{.Date.Format "2006-01-02"}} | <a href="{{.RelPermalink}}">{{.Title}}</a></li>
+	{{end}}
+</ul>
+
+<!--pagination-->
+{{template "_internal/pagination.html" .}}
+
+{{end}}
```
Comment (galargh) [inline www/themes/sk1/layouts/_default/list.html:25]:
```suggestion
{{end}}

```

File: www/themes/sk1/layouts/_default/baseof.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+<body>
+	{{- partial "header.html" . -}}
+	<div id="content">
+		{{- block "main" . }}{{- end }}
+	</div>
+	{{- partial "footer.html" . -}}
+</body>
+
+</html>
```
Comment (galargh) [inline www/themes/sk1/layouts/_default/baseof.html:13]:
```suggestion
</html>

```

File: www/themes/sk1/layouts/404.html [hunk 11]
```diff
@@ -0,0 +1,3 @@
+{{define "main"}}
+Page Not Found.
+{{end}}
```
Comment (galargh) [inline www/themes/sk1/layouts/404.html:3]:
```suggestion
{{end}}

```

**Review by galargh — APPROVED** [review]

File: .github/workflows/deploy-pages.yml (hunk 2 above)
Comment (laurentsenta) [reply .github/workflows/deploy-pages.yml:82]: We don't right now, but it's quite common to use submodule with hugo themes

File: www/static/favicon.ico
Comment (laurentsenta) [reply www/static/favicon.ico:1]: Yea, it disables the one define in sk1 theme.

File: www/themes/sk1/LICENSE
Comment (laurentsenta) [reply www/themes/sk1/LICENSE:1]: I've seen a few different approaches, submodule / subtrees, copies, and hugo modules. Example in the [theme's doc itself](https://github.com/J-Siu/hugo-theme-sk1#install)

In that case, we're using such a minimal theme, I expect to replace all the components in one or two PR. We'll remove the folder eventually so cloning in the theme folder made sense here.

File: www/static/logo.png
Comment (laurentsenta) [reply www/static/logo.png:1]: midjourney 🤖

**Pull request merged**